build/build.py

```diff
@@ -434,7 +435,11 @@ def platform(self):
 
     @property
     def generator(self):
-        return 'Ninja'
+        return self._generator or 'Ninja'
+
+    @generator.setter
+    def generator(self, value):
```

Comment:
>generator [](http://example.com/codeflow?start=8&length=9)

Instead of repeating the setter in derived classes, can you move it to the base class BuildRunnerBase?

Reply:
I could but then I would still have to have the derived call settings there to call the parents setter which results in more code.  
`    @generator.setter
    def generator(self, value):
        super(LinuxBuildRunner, type(self)).generator.fset(self, value)`

If that is what you want just let me know and I will make the changes